Lib/http/server.py

```diff
@@ -686,10 +687,30 @@ def send_head(self):
         except OSError:
             self.send_error(HTTPStatus.NOT_FOUND, "File not found")
             return None
+
+        fs = os.fstat(f.fileno())
+            
         try:
+            # Use browser cache if possible
+            if ("If-Modified-Since" in self.headers
+                    and "If-None-Match" not in self.headers):
+                # compare If-Modified-Since and time of last file modification
+                ims = email.utils.parsedate_to_datetime(
+                    self.headers["If-Modified-Since"])
+                if (ims is not None and
+                        ims.tzinfo is datetime.timezone.utc):
+                    # compare to UTC datetime of last modification, 
+                    # rounded to the second
+                    mtime = int(fs.st_mtime)
+                    last_modif = datetime.datetime.fromtimestamp(mtime, 
+                        ims.tzinfo)
+                    if last_modif <= ims:
+                        self.send_response(HTTPStatus.NOT_MODIFIED)
+                        self.end_headers()
+                        f.close()
+                        return
```

Comment:
`return None`. If there are `return` statements with value, all `None` values should be returned explicitly.